packages/voice/src/util/Secretbox.ts

```diff
@@ -1,70 +1,108 @@
-		open: sodium.api.crypto_secretbox_open_easy,
-		close: sodium.api.crypto_secretbox_easy,
-		random: (num: number, buffer: Buffer = Buffer.allocUnsafe(num)) => {
-			sodium.api.randombytes_buf(buffer);
-			return buffer;
+		crypto_aead_xchacha20poly1305_ietf_decrypt: (
+			cipherText: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.api.crypto_aead_xchacha20poly1305_ietf_decrypt(cipherText, additionalData, null, nonce, key);
+		},
+		crypto_aead_xchacha20poly1305_ietf_encrypt: (
+			plaintext: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.api.crypto_aead_xchacha20poly1305_ietf_encrypt(plaintext, additionalData, null, nonce, key);
 		},
 	}),
 	'libsodium-wrappers': (sodium: any): Methods => ({
-		open: sodium.crypto_secretbox_open_easy,
-		close: sodium.crypto_secretbox_easy,
-		random: sodium.randombytes_buf,
+		crypto_aead_xchacha20poly1305_ietf_decrypt: (
+			cipherText: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.crypto_aead_xchacha20poly1305_ietf_decrypt(null, cipherText, additionalData, nonce, key);
+		},
+		crypto_aead_xchacha20poly1305_ietf_encrypt: (
+			plaintext: Buffer,
+			additionalData: Buffer,
+			nonce: Buffer,
+			key: ArrayBufferLike,
+		) => {
+			return sodium.crypto_aead_xchacha20poly1305_ietf_encrypt(plaintext, additionalData, null, nonce, key);
+		},
 	}),
-	tweetnacl: (tweetnacl: any): Methods => ({
-		open: tweetnacl.secretbox.open,
-		close: tweetnacl.secretbox,
-		random: tweetnacl.randomBytes,
+	'@stablelib/xchacha20poly1305': (stablelib: any): Methods => ({
+		crypto_aead_xchacha20poly1305_ietf_decrypt(plaintext, additionalData, nonce, key) {
+			const crypto = new stablelib.XChaCha20Poly1305(key);
+			return crypto.open(nonce, plaintext, additionalData);
+		},
+		crypto_aead_xchacha20poly1305_ietf_encrypt(cipherText, additionalData, nonce, key) {
+			const crypto = new stablelib.XChaCha20Poly1305(key);
+			return crypto.seal(nonce, cipherText, additionalData);
+		},
 	}),
 } as const;
 
 const fallbackError = () => {
 	throw new Error(
 		`Cannot play audio as no valid encryption package is installed.
-- Install sodium, libsodium-wrappers, or tweetnacl.
+- Install sodium, libsodium-wrappers, or @stablelib/xchacha20poly1305.
 - Use the generateDependencyReport() function for more information.\n`,
 	);
 };
 
 const methods: Methods = {
-	open: fallbackError,
-	close: fallbackError,
-	random: fallbackError,
+	crypto_aead_xchacha20poly1305_ietf_encrypt: fallbackError,
+	crypto_aead_xchacha20poly1305_ietf_decrypt: fallbackError,
 };
 
 void (async () => {
 	for (const libName of Object.keys(libs) as (keyof typeof libs)[]) {
 		try {
-			// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
-			const lib = require(libName);
+			const lib = await import(libName);
```

Comment:
Any particular reason for this change? Do we have any way to _not_ require this to become async (as it can come with its own can of worms)... Maybe we should make this secretbox module a lazy loaded thing thats awaitable (export the two functions that handle imports themselves the first time)

Reply:
You can create the `require` function [in ESM] by using [`createRequire()`](https://nodejs.org/api/module.html#modulecreaterequirefilename) from `node:module`

Reply:
Realistically its still something we ought to change, this random `void Promise` thing is hella janky imo (and prone to race conditions or worse)

Reply:
I'm for a `lazy` top level.

Reply:
Would it cause issues to export en/decrypt methods top level that are async and that handle the imports themselves? 👀 (or that are sync and export an `initCrypto` or w/e function that gets called first thing on connect, assuming then that connect is async)

Alternatively similar to what dd said (which we already do in /ws for zlib-sync)

Reply:
Was the move to `import` required? Otherwise we should probably leave it be as is unless others have a better idea

Reply:
I mean you didn't show what the alternative is 😅, so 🤷‍♂️

Reply:
Sure! But would love the thoughts of @didinele and/or @discordjs/core about this too (if we should just leave it as is or not)

Reply:
[`@noble/ciphers`](https://www.npmjs.com/package/@noble/ciphers) looks good to me 👍🏼

Reply:
I'll take a look at helping out with this 🫡

Reply:
Shouldn’t this be reverted to require then (since that was the whole reason for choosing that lib)?